src/lib/resolvers/message.ts

```diff
@@ -1,28 +1,67 @@
 import { ChannelMessageRegex, MessageLinkRegex, SnowflakeRegex } from '@sapphire/discord-utilities';
-import { GuildBasedChannelTypes, isNewsChannel, isTextChannel, TextBasedChannelTypes } from '@sapphire/discord.js-utilities';
+import {
+	GuildBasedChannelTypes,
+	isGuildBasedChannel,
+	isNewsChannel,
+	isTextBasedChannel,
+	isTextChannel,
+	TextBasedChannelTypes
+} from '@sapphire/discord.js-utilities';
 import { container } from '@sapphire/pieces';
 import { err, ok, Result } from '@sapphire/result';
 import type { Awaitable } from '@sapphire/utilities';
 import { Message, Permissions, Snowflake, User } from 'discord.js';
 import { Identifiers } from '../errors/Identifiers';
 
-interface MessageResolverOptions {
+/**
+ * Options to resolve a message from a string, given a certain context.
+ */
+export interface MessageResolverOptions {
+	/**
+	 * Channel to resolve the message in.
+	 * @default message.channel
+	 */
 	channel?: TextBasedChannelTypes;
+	/**
+	 * Base message to resolve the message from (e.g. pick the channel if not given).
+	 */
 	message: Message;
+	/**
+	 * Whether to scan the entire guild cache for the message.
+	 * If channel is given with this option, this option is ignored.
+	 * @default false
+	 */
+	scan?: boolean;
 }
 
 export async function resolveMessage(parameter: string, options: MessageResolverOptions): Promise<Result<Message, Identifiers.ArgumentMessageError>> {
-	const channel = options.channel ?? options.message.channel;
 	const message =
-		(await resolveById(parameter, channel)) ??
+		(await resolveById(parameter, options)) ??
 		(await resolveByLink(parameter, options.message)) ??
 		(await resolveByChannelAndMessage(parameter, options.message));
 	if (message) return ok(message);
 	return err(Identifiers.ArgumentMessageError);
 }
 
-function resolveById(parameter: string, channel: TextBasedChannelTypes): Awaitable<Message | null> {
-	return SnowflakeRegex.test(parameter) ? channel.messages.fetch(parameter as Snowflake) : null;
+function resolveById(parameter: string, options: MessageResolverOptions): Awaitable<Message | null> {
+	if (!SnowflakeRegex.test(parameter)) return null;
+
+	if (options.channel) {
+		const channel = options.channel ?? options.message.channel;
```

Comment:
This will never run if channel is undefined, so line 50 is useless.. Might want to change logic up

Reply:
How did I not see that...
Thanks, I'll fix it in the afternoon

Reply:
yep that's a long afternoon 👀 
sorry I'll do that this Sunday, something came up.

Reply:
Done in b5ec37d

Reply:
Done in b5ec37d